plugins/local/non_h_ints_mu/jast_1e_utils.irp.f

```diff
@@ -231,36 +250,69 @@ subroutine get_j1e_coef_fit_ao2(dim_fit, coef_fit)
   ! --- --- ---
   ! solve Ax = b
 
-  allocate(A_inv(ao_num*ao_num,ao_num*ao_num))
-  !call get_inverse(A, ao_num*ao_num, ao_num*ao_num, A_inv, ao_num*ao_num)
-  call get_pseudo_inverse(A, ao_num*ao_num, ao_num*ao_num, ao_num*ao_num, A_inv, ao_num*ao_num, 5d-8)
+  !call get_pseudo_inverse(A, ao_num*ao_num, ao_num*ao_num, ao_num*ao_num, A_inv, ao_num*ao_num, cutoff_svd)
 
-  ! coef_fit = A_inv x b
-  call dgemv("N", ao_num*ao_num, ao_num*ao_num, 1.d0, A_inv, ao_num*ao_num, b, 1, 0.d0, coef_fit, 1)
+  allocate(D(ao_num*ao_num), U(ao_num*ao_num,ao_num*ao_num), Vt(ao_num*ao_num,ao_num*ao_num))
 
-  integer          :: mn
-  double precision :: tmp, acc, nrm
+  allocate(work(1))
+  lwork = -1
+  call dgesvd( 'S', 'A', ao_num*ao_num, ao_num*ao_num, A(1,1,1,1), ao_num*ao_num &
+             , D(1), U(1,1), ao_num*ao_num, Vt(1,1), ao_num*ao_num, work, lwork, info)
+  if(info /= 0) then
+    print *,  info, ': SVD failed'
+    stop
+  endif
 
-  acc = 0.d0
-  nrm = 0.d0
+  LWORK = max(5*ao_num*ao_num, int(WORK(1)))
+  deallocate(work)
+  allocate(work(lwork))
+  call dgesvd( 'S', 'A', ao_num*ao_num, ao_num*ao_num, A(1,1,1,1), ao_num*ao_num &
+             , D(1), U(1,1), ao_num*ao_num, Vt(1,1), ao_num*ao_num, work, lwork, info)
+  if(info /= 0) then
+    print *,  info, ':: SVD failed'
+    stop 1
+  endif
+
+  deallocate(work)
+
+  n_svd = 0
   do ij = 1, ao_num*ao_num
-    tmp = 0.d0
-    do kl = 1, ao_num*ao_num
-      tmp += A(ij,kl) * coef_fit(kl)
-    enddo
-    tmp = tmp - b(ij)
-    if(dabs(tmp) .gt. 1d-7) then
-      print*, ' problem found in fitting 1e-Jastrow'
-      print*, ij, tmp
+    if(D(ij)/D(1) > cutoff_svd) then
```

Comment:
Check that `D(1) != 0.d0` here to avoid NaN.

Reply:
Ok